plugins/calendar/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarLogicTest.java

```diff
@@ -57,17 +60,57 @@
 	private TestFramework framework;
 
 	private CalendarLogic callog;
+	private CalendarService calendarService;
 
 	/**
 	 * Initializes the class variables before each test
 	 */
 	@BeforeEach
 	public void setup() {
-		this.framework.mockService(CalendarService.class);
+		this.calendarService = this.framework.mockService(CalendarService.class);
 		this.framework.mockService(Environment.class);
 		this.callog = this.framework.setServiceUnderTest(CalendarLogic.class);
 	}
 
+	@ParameterizedTest
+	@MethodSource("testSetEvent")
+	public void testCheckSetEvent(CalendarEvent calendarEvent) {
+		this.callog.setEvent(calendarEvent);
+		boolean allDay = calendarEvent.isAllDay();
+		Mockito.verify(this.calendarService).addEvent(ArgumentMatchers.eq("primary"),
+				ArgumentMatchers.argThat(event -> {
+					return event.getSummary().equals(calendarEvent.getSummary())
+							&& event.getLocation().equals(calendarEvent.getLocation())
+							&& event.getDescription().equals(calendarEvent.getDescription())
+							&& ((event.getStart().getDate() != null) == allDay) && allDay
+									? (event.getStart().getDate().equals(
+											EDTfromLocalDateTime(calendarEvent.getStart(), allDay).getDate())
+											&& event.getEnd().getDate()
+													.equals(EDTfromLocalDateTime(calendarEvent.getEnd(), allDay)
+															.getDate()))
+									: (event.getStart().getDateTime().equals(
+											EDTfromLocalDateTime(calendarEvent.getStart(), allDay).getDateTime())
+											&& event.getEnd().getDateTime().equals(
+													EDTfromLocalDateTime(calendarEvent.getEnd(), allDay).getDateTime()))
+											&& event.getRecurrence()
+													.equals(Arrays.asList(calendarEvent.getRecurrence()))
+											&& event.getReminders().equals(calendarEvent.getReminders());
+				}));
```

Comment:
Simplify please. Also, if you test with branch coverage, it says that only 1 out of 2 branches is covered here